Sources/AblyAssetTrackingInternal/WorkerQueue/Worker.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import AblyAssetTrackingCore
+
+/// A [Worker] interface represents workers which execute work inside [WorkerQueue].
+/// PropertiesType: - the type of properties used by this worker as both input and output.
+/// WorkerSpecificationType - the type of specification used to post another worker back to the queue.
+public protocol Worker<PropertiesType, WorkerSpecificationType>: AnyObject {
```

Comment:
That's how I tried approaching it in the beginning, but I ran into some problems with limitations of Swift's generics along the way. I'll have another look though.

Reply:
Can't remember what the problem was back when I tackled it last time, maybe some Xcode shenanigans, but it worked this time. Added CallbackWorker and DefaultWorker protocols with default impl in 76e4ba2fa233a831d8ede80d837790a6e90ecd6c

I'm now wondering though - shouldn't we be operating there already on `ErrorInformation`s instead of on `Error`s?

Reply:
Yeah, please disregard. I was wondering whether the whole WorkerQueue could operate on ErrorInformation, but it's not realistic.